src/components/RenderRouter/ArchiveItem.tsx

```diff
@@ -0,0 +1,269 @@
+
+import React, { EventHandler, SyntheticEvent } from 'react';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+import "./ArchiveItem.scss"
+import Amplify from 'aws-amplify';
+import awsmobile from '../../aws-exports';
+import DataLoader, {
+    VideoByVideoTypeData,
+    SeriesByTypeData
+} from './DataLoader';
+import VideoOverlay from '../VideoOverlay/VideoOverlay';
+import format from 'date-fns/format';
+
+Amplify.configure(awsmobile);
+
+type listData = SeriesByTypeData | VideoByVideoTypeData
+
+interface Props extends RouteComponentProps {
+    content: any,
+    data: any,
+    pageConfig: any,
+}
+interface State {
+    listData: listData[]
+    overlayData: any
+    urlHistoryState: any
+    selectedYear: string
+    yearOptions: string[]
+    currentNextToken: string | null
+}
+class ArchiveItem extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            listData: [],
+            overlayData: null,
+            urlHistoryState: window.history.state,
+            selectedYear: 'All',
+            yearOptions: ['All'],
+            currentNextToken: 'placeholder'
+        }
+    }
+
+    async componentDidMount() {
+        const dataLoaded = (data: listData[]) => this.setData(data);
+        const checkNext = (data: string | null) => this.setState({ currentNextToken: data })
```

Comment:
This function isn't checking really anything, so the name is misleading. Maybe "tokenCallback"? But you could also pass this to the "dataLoaded" function, since the next token is part of the data that has been returned.